### PR TITLE
Make severity optional in Problems API

### PR DESCRIPTION
**Pull request by reinsch82**
### Reviewing cheatsheet

Before merging the PR, comments starting with 
- ❌ ❓**must** be fixed
- 🤔 💅 **should** be fixed
- 💭 **may** be fixed
- 🎉 celebrate happy things

### CODE REVIEW

File: subprojects/plugin-development/src/main/java/org/gradle/plugin/devel/tasks/internal/ValidateAction.java [hunk 1]
```diff
@@ -73,7 +73,7 @@ public interface Params extends WorkParameters {
 
     }
 
-//    @Inject
+    //    @Inject
 //    protected abstract Problems getProblemsService();
```
Comment (donat) [inline subprojects/plugin-development/src/main/java/org/gradle/plugin/devel/tasks/internal/ValidateAction.java:77]: This seems like an accidental change.
Should we remove commented lines?

File: subprojects/problems/src/main/java/org/gradle/api/problems/Problems.java [hunk 2]
```diff
@@ -39,11 +39,24 @@
 @ServiceScope(Scope.Global.class)
 public abstract class Problems {
 
+    /**
+     * A function that can be used to configure a {@link ProblemBuilder}.
+     *
+     * @since 8.4
+     */
+    @Incubating
+    public interface ProblemConfigurator {
```
Comment (donat) [inline subprojects/problems/src/main/java/org/gradle/api/problems/Problems.java:48]: ❌ I think this is unnecessary.

The throwing and rethrowing could be using an Action as a parameter:
```
RuntimeException throwing(
    Action<ProblemBuilderDefiningDocumentation> problemSpec
);

RuntimeException rethrowing(
    RuntimeException e, 
    Action<ProblemBuilderDefiningDocumentation> problemSpec
);
```

The implementation will only need an extra line as  the action doesn't return anything:
```
    public RuntimeException throwing(Action<ProblemBuilderDefiningDocumentation> problemSpec) {
        DefaultProblemBuilder defaultProblemBuilder = createProblemBuilderInternal();
        problemSpec.execute(defaultProblemBuilder);
        throw defaultProblemBuilder.throwIt();
    }
```

**Review by donat — CHANGES_REQUESTED** [review]

File: subprojects/problems/src/main/java/org/gradle/api/problems/Problems.java (hunk 2 above)
Comment (reinsch82) [reply subprojects/problems/src/main/java/org/gradle/api/problems/Problems.java:48]: By using a function style parameter we can force the user to go through the all the required properties.
Don't you think that adds value?

File: subprojects/plugin-development/src/main/java/org/gradle/plugin/devel/tasks/internal/ValidateAction.java (hunk 1 above)
Comment (reinsch82) [reply subprojects/plugin-development/src/main/java/org/gradle/plugin/devel/tasks/internal/ValidateAction.java:77]: already in the works on the next PR :)

**Review by donat — APPROVED** [review]

**Comment by reinsch82** [comment]
@bot-gradle test and merge

**Pull request merged**